WcaOnRails/app/controllers/api/v0/roles_controller.rb

```diff
@@ -8,53 +8,95 @@ class Api::V0::RolesController < Api::V0::ApiController
     end
   end
 
+  # Returns a list of roles based on the request parameters.
```

Comment:
I am not a big fan of having one route that does many things based on which query params you pass in.
Is there any chance you can create separate GET endpoints for the separate "types" of data?

Reply:
(I'll attend to the other messages soon, but thought of attending this immediately as I am not clear how to proceed)

Currently, `/api/v0/roles` is returning the list of roles based on parameters. One of the following three parameters must be there:
- userId
- groupId
- groupType

I'm guessing what you meant is to have separate routes for each of the above three: like `api/v0/user-id-roles`, `api/v0/group-id-roles` and `api/v0/group-type-roles`.

But I'm also confused whether by 'types' you meant `groupType` and hence we will have endpoints like: `api/v0/delegate-regions-roles`, `api/v0/teams-roles`, `api/v0/councils-roles`, etc.

I hope you meant the first one, can you please clarify this?

Reply:
Yes, I meant the first one. When I spoke of types, I explicitly mentioned "types of data", **not** the types of groups.

My suggestion would be the following, although this is not strict and you are welcome to make any changes as you see fit
- GET `api/v0/roles/user/{userId}`
- GET `api/v0/roles/{groupId}`
- GET `api/vo/roles/type/{groupType}`